notifications/src/main/java/org/openedx/notifications/presentation/settings/NotificationsSettingsViewModel.kt

```diff
@@ -23,39 +32,72 @@ class NotificationsSettingsViewModel(
     val notificationsConfiguration: StateFlow<NotificationsConfiguration>
         get() = _notificationsConfiguration
 
+    private val _showPermissionRequestDialog = MutableStateFlow(false)
+    val showPermissionRequestDialog: StateFlow<Boolean>
+        get() = _showPermissionRequestDialog
+
+    private val _uiMessage = MutableSharedFlow<UIMessage>()
+    val uiMessage = _uiMessage.asSharedFlow()
+
     init {
-        viewModelScope.launch {
+        if (checkPushNotificationPermission()) {
             fetchAndUpdateNotificationsSettings()
+        } else {
+            updateDiscussionPreference(updatedValue = false)
+            _showPermissionRequestDialog.update { true }
```

Comment:
The following dialog should be displayed when the user try to interact with switch.
<img src="https://github.com/user-attachments/assets/a23c5729-d6ca-4bbb-9790-01e495f39cdb" height="512"/>